src/urban_disco/core.clj

```diff
@@ -1,51 +1,8 @@
 (ns urban-disco.core
+  (require [grid.base-grid :as grid-builder]
+           [grid.tile :as tile])
   (:gen-class))
 
-(defn build-tile [x y grid-height] 
-  {:x        x
-   :y        y
-   :explored false
-   :up       (if (= y 0) :wall :door)
-   :down     (if (= (+ y 1) grid-height) :wall :door)
-   :left     (if (= x 0) :wall :door)
-   :right    (if (= (+ x 1) grid-height) :wall :door)})
-
-(defn build-row [width row-number]
-  (let [row (range width)]
-    (into [] (map #(build-tile % row-number width) row))))
-
-(defn build-grid
-  ([] (build-grid 5))
-  ([height] 
-   (let [grid (range height)]
-     (into [] (map #(build-row height %) grid)))))
-
-(defn get-tile [grid x y]
-  (nth (nth grid y) x))
-
-(defn calc-center [height]
-  (-> height dec (/ 2) Math/ceil int))
-
-(defn calc-vector-center [grid]
-  (-> grid count calc-center))
-
-(defn grid-center [grid]
-  (let [x (calc-vector-center grid)
-        y (calc-vector-center (first grid))]
-    (get-tile grid x y)))
-
-(defn calc-starting-tile
-  ([] (calc-starting-tile 5))
-  ([height]
-   (let [center (calc-center height)]
-     {:x center :y center})))
-
-(defn is-tile-explored [tile]
-  (get tile :explored))
-
-(defn explore-tile [state x y]
```

Comment:
Since this is gone, the calls to explore tile in all of the `move-*` functions below no longer make sense.

Reply:
moving code needs to be rewritten to support 1 dimensional vectors.
Will rewrite and move into its own module.